codegen/codegen.c

```diff
@@ -84,6 +87,56 @@ char *new_label(context *ctx)
     return name;
 }
 
+char *new_loop_end_label(context *ctx)
+{
+    char *name = new_label(ctx);
+    int *label_id = malloc(sizeof(int));
+    *label_id = ctx->label_counter-1;
+    add_to_list(ctx->loop_end_labels, label_id);
+    return name;
+}
+
+char* get_current_loop_end_label_counter(context *ctx, char *name)
+{
+    if (ctx->loop_end_labels->count == 0) {
+        fprintf(stderr, "Not inside loop\n");
+        return NULL;
+    }
+    int *value = (int *)ctx->loop_end_labels->last->value;
+    sprintf(name, "__tmp_label_%u", *value);
+    return name;
+}
+
+char *new_loop_start_label(context *ctx)
+{
+    char *name = new_label(ctx);
+    int *label_id = malloc(sizeof(int));
+    *label_id = ctx->label_counter-1;
+    add_to_list(ctx->loop_start_labels, label_id);
+    return name;
+}
+
+char* get_current_loop_start_label_counter(context *ctx, char *name)
+{
+    if (ctx->loop_start_labels->count == 0) {
+        fprintf(stderr, "Not inside loop\n");
+        return NULL;
+    }
+    int *value = (int *)ctx->loop_start_labels->last->value;
+    sprintf(name, "__tmp_label_%u", *value);
+    return name;
+}
+
+void exit_loop(context *ctx)
+{
+    if (ctx->loop_end_labels->count > 0)
+        pop_list(ctx->loop_end_labels);
+
+    if (ctx->loop_start_labels->count > 0)
+        pop_list(ctx->loop_start_labels);
+    add_text(ctx, "; exit loop");
```

Comment:
I'm thinking to have these pointers in the assembly code to help with reviewing them. Not sure if this is the best way.

Reply:
It's fine to have comments in the asm output, it's so useful for debugging purposes